### PR TITLE
Remove build tool from bootstrapped applications

### PR DESCRIPTION
**Pull request by ZachEddy**
# Remove Build Tool

This pull request removes the Infoblox build tool from boostrapped applications in favor of multi-stage Dockerfiles. In my opinion, this approach makes the bootstrapped Makefile much cleaner and easier to understand.

[Here's](https://github.com/ZachEddy/atlas-app-toolkit/tree/remove-buildtool-example/cli/example) an example project that doesn't use the build tool. The Makefiles and Dockerfiles are the points of interest in this example.

### CODE REVIEW

File: cli/atlas/templates/Makefile.gotmpl [hunk 1]
```diff
@@ -1,75 +1,61 @@
...
 {{ if .WithGateway }}
-.PHONY: gateway
-gateway: gateway-build gateway-docker
-	@$(BUILDER) go build $(GO_BUILD_FLAGS) -o $(SERVER_BINARY) $(SERVER_PATH)
-
-.PHONY: gateway-build
+.PHONY:
 gateway-build:
-	@$(BUILDER) go build $(GO_BUILD_FLAGS) -o $(GATEWAY_BINARY) $(GATEWAY_PATH)
+	@go build $(GO_BUILD_FLAGS) -o $(GATEWAY_BINARY) $(GATEWAY_PATH)
```
[the remark this answers is not in the document]
Comment (ZachEddy) [reply cli/atlas/templates/Makefile.gotmpl]: I think the `gateway-build` and `server-build` make targets can be removed completely. The building of Go binaries occurs in the Dockerfile.

**Comment by ZachEddy** [comment]
Ping.

File: cli/atlas/templates/Makefile.gotmpl (hunk 1 above)
[the remark this answers is not in the document]
Comment (dmacthedestroyer) [reply cli/atlas/templates/Makefile.gotmpl]: let's just remove these targets, then..

**Comment by ZachEddy** [comment]
I updated [the example application](https://github.com/ZachEddy/atlas-app-toolkit/tree/remove-buildtool-example/cli/example).

File: cli/atlas/templates/Makefile.gotmpl [hunk 2]
```diff
@@ -1,76 +1,53 @@
...
-	
-	DOCKER_RUNNER        	:= docker run --rm
-	DOCKER_RUNNER        	+= -v $(SRCROOT_ON_HOST):$(SRCROOT_IN_CONTAINER)
-	DOCKER_BUILDER       	:= infoblox/buildtool:v8
-	DOCKER_GENERATOR     	:= infobloxcto/gentool:v1
-	BUILDER              	:= $(DOCKER_RUNNER) -w $(SRCROOT_IN_CONTAINER) $(DOCKER_BUILDER)
-	GENERATOR            	:= $(DOCKER_RUNNER) $(DOCKER_GENERATOR)
-endif
+# configuration for the protobuf gentool
+SRCROOT_ON_HOST		:= $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
```
Comment (johnbelamaric) [inline cli/atlas/templates/Makefile.gotmpl]: Why is the spacing/tab different here?

File: cli/atlas/templates/Makefile.gotmpl [hunk 3]
```diff
@@ -1,76 +1,53 @@
-PROJECT_ROOT 				:= {{ if .Root }}{{ .Root }}/{{ .Name }}{{ else }}{{ .Name }}{{ end }}
-BUILD_PATH  				:= bin
-DOCKERFILE_PATH 		:= $(CURDIR)/docker
+PROJECT_ROOT		:= {{ if .Root }}{{ .Root }}/{{ .Name }}{{ else }}{{ .Name }}{{ end }}
+BUILD_PATH		:= bin
+DOCKERFILE_PATH		:= $(CURDIR)/docker
```
Comment (johnbelamaric) [inline cli/atlas/templates/Makefile.gotmpl]: Why is the spacing/tab different here?

Comment (ZachEddy) [reply cli/atlas/templates/Makefile.gotmpl]: I think the git diff is causing indentation issues, but I'm not sure why. The [template itself](https://github.com/ZachEddy/atlas-app-toolkit/blob/1e1677af0e40fcc28b2288b3fb2dc37e4b255326/cli/atlas/templates/Makefile.gotmpl) doesn't have spacing/tab problems and it renders [Makefiles](https://github.com/ZachEddy/atlas-app-toolkit/blob/remove-buildtool-example/cli/example/Makefile) that look okay.

**Pull request merged**